src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+    .spyOn(global.console, "error")
+    .mockImplementation(() => undefined);
+  const expected =
+    "Forbidden prop dateFormat supplied to NumeralDate. " +
+    "Only one of these date formats is allowed: " +
+    "['dd', 'mm', 'yyyy'], ['mm', 'dd', 'yyyy'], ['yyyy', 'mm', 'dd'], ['dd', 'mm'], ['mm', 'dd'], ['mm', 'yyyy']";
+
+  expect(() =>
+    render(
+      <NumeralDate
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore:next-line testing incorrect date format
+        dateFormat={["xx"]}
+        value={{ dd: "", mm: "", yyyy: "" }}
+        onChange={() => {}}
+      />
+    )
+  ).toThrow(expected);
+  consoleSpy.mockReset();
+});
+
+test("should not throw an error if no `dateFormat` is passed", () => {
+  const consoleSpy = jest
+    .spyOn(global.console, "error")
+    .mockImplementation(() => undefined);
+  render(
+    <NumeralDate value={{ dd: "", mm: "", yyyy: "" }} onChange={() => {}} />
+  );
+
+  expect(consoleSpy).not.toHaveBeenCalled();
+  consoleSpy.mockReset();
+});
+
+test("should throw when component changes from uncontrolled to controlled", () => {
+  const consoleSpy = jest
+    .spyOn(global.console, "error")
+    .mockImplementation(() => undefined);
+  const { rerender } = render(<NumeralDate value={undefined} />);
+  expect(() => {
+    rerender(<NumeralDate value={{ dd: "02", mm: "01", yyyy: "2020" }} />);
+  }).toThrow(
+    "Input elements should not switch from uncontrolled to controlled (or vice versa). " +
+      "Decide between using a controlled or uncontrolled input element for the lifetime of the component"
+  );
+  consoleSpy.mockReset();
+});
+
+describe("when the `error` prop is passed a string value and `validationRedesignOptIn` is not set", () => {
+  it("should render with `aria-invalid` set to 'true'", () => {
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "02", mm: "01", yyyy: "2020" }}
+          onChange={() => {}}
+          error="error"
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+  });
+
+  it("should display the tooltip when the user hovers on the 'Day' input", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider>
+        <NumeralDate
+          value={{ dd: "02", mm: "01", yyyy: "2020" }}
+          onChange={() => {}}
+          error="error"
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    await user.hover(dayInput);
+
+    expect(screen.getByRole("tooltip", { name: "error" })).toBeVisible();
```

Comment:
**suggestion** (non-blocking): since this is the string case, it might be worth also checking that the tooltip has the expected text content (with `.toHaveTextContent("error")`

(Similarly for the other tooltip tests with a string validation prop, but I won't repeat myself for this one.)

Reply:
I believe this is the approach we agreed as preferred, using the `name` property is essentially selecting it via the text and then asserting it as visible etc. I'd prefer to leave these as is for now as overall I think we've adopted this same approach in other files (we do have the approach you mention as well though) so if we decide to pivot I'd prefer to do them all at once